### DISCUSSION TITLE
loading s3 image url format issues.

### DISCUSSION THREAD

**Discussion by Auggie321** [gethinode/hinode#554] [Help] [answered]
sir,

with hinode v0.20.7, check out there [components/image](https://gethinode.com/docs/components/image/)
```markdown
## example1.jpg set picture size and centerd it in markdown
<div align="center"> <img src="https://s3/af3dc777d10709d08ba8813.jpg" alt="example2.jpg" width="70%" height="70%"> </div>

## multiple jpg
|<img src="https://s3/agmx-img/2023/8ba8813.jpg" width="200" />|<img src="https://s3/agmx-img/2023/8ba8813.jpg" width="200" />|
|-|-
```
tried some ways, but failed.
may i have ask the question that how to use correct format with your template markdown ?
thanks.

**Comment by markdumay** [comment, marked as answer]
If I understand your request correctly, you're trying to include an image hosted on Amazon S3 in your website. From the docs:

>Use the image shortcode to display a responsive image with a specific aspect ratio. The source link can refer to either an image available in the /assets/img folder of your site or a **public web location**. The shortcode renders the image as a so-called [image set ](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images) to optimize the image for different screen sizes and resolutions. Behind the scenes, Hugo renders the images in WebP format and stores them in a local folder (resources or public). The images are processed using the quality setting specified in the [imaging] section of the main [config file ](https://gohugo.io/content-management/image-processing/#imaging-configuration)(defaults to 75). Supported image types are .png, .jpeg, .gif, .tiff, .bmp, and .webp. A fallback image of type .jpeg is provided for older browsers.

In other words, you can reference an external image as long at it is **publicly accessible**. The following shortcode is a working example:

```
{{< image src="https://gethinode.com/img/puzzle-1400x600.webp" >}}
```

The image source locations in your example point to a S3 bucket. The S3 bucket needs to be publicly accessible and requires a valid URL of the format `https://s3.amazonaws/bucket_name/key_name/image_name.jpg`.  Hinode downloads and processes this image during build, so the final image set is always hosted on your own webserver.

Stack Overflow provides more [background about referencing S3 images](https://stackoverflow.com/questions/27768596/how-do-i-show-an-image-from-my-amazon-s3-on-my-website).

**Reply by Auggie321** [reply]
Thank you very much for your response. 
After a careful review and comparison with testing, 
I've come to understand that the current theme still tends to save blog images in the 'assess/img' directory and relies on server-side compilation and cropping during page display. This approach ensures the best visual layout when the site is accessed from multiple client devices such as smartphones and computers. I noticed there are some restrictions on retrieving remote URL images in the Getimage.html file.

Although standard markdown format allows for simple referencing of online images from cloudflare R2 bucket public, using this format directly on mobile or desktop results in poor display and aesthetics, as confirmed during practical testing.

No wonder the documentation doesn't provide instructions on the standard markdown format for cropping online images. 
It's clear to me now, and the problem has been resolved. 
Thank you very much.
